src/services/identity/SitesService.ts

```diff
@@ -1,46 +1,207 @@
+import Bluebird from "bluebird"
+import _ from "lodash"
 import { ModelStatic } from "sequelize"
 
-import { Site } from "@database/models"
+import { Site, SiteMember, User } from "@database/models"
+import UserSessionData from "@root/classes/UserSessionData"
+import {
+  ISOMER_GITHUB_ORG_NAME,
+  ISOMERPAGES_REPO_PAGE_COUNT,
+  GH_MAX_REPO_COUNT,
+  ISOMER_ADMIN_REPOS,
+} from "@root/constants"
+import { NotFoundError } from "@root/errors/NotFoundError"
+import { genericGitHubAxiosInstance } from "@root/services/api/AxiosInstance"
+import type { GitHubRepositoryData, RepositoryData } from "@root/types/repoInfo"
+import { GitHubService } from "@services/db/GitHubService"
+import { ConfigYmlService } from "@services/fileServices/YmlFileServices/ConfigYmlService"
+import IsomerAdminsService from "@services/identity/IsomerAdminsService"
+import UsersService from "@services/identity/UsersService"
 
 import TokenStore from "./TokenStore"
 
 interface SitesServiceProps {
-  repository: ModelStatic<Site>
+  siteRepository: ModelStatic<Site>
+  gitHubService: GitHubService
+  configYmlService: ConfigYmlService
+  usersService: UsersService
+  isomerAdminsService: IsomerAdminsService
   tokenStore: TokenStore
 }
 
 class SitesService {
   // NOTE: Explicitly specifying using keyed properties to ensure
   // that the types are synced.
-  private readonly repository: SitesServiceProps["repository"]
+  private readonly siteRepository: SitesServiceProps["siteRepository"]
+
+  private readonly gitHubService: SitesServiceProps["gitHubService"]
+
+  private readonly configYmlService: SitesServiceProps["configYmlService"]
+
+  private readonly usersService: SitesServiceProps["usersService"]
+
+  private readonly isomerAdminsService: SitesServiceProps["isomerAdminsService"]
 
   private readonly tokenStore: SitesServiceProps["tokenStore"]
 
-  constructor({ repository, tokenStore }: SitesServiceProps) {
-    this.repository = repository
+  constructor({
+    siteRepository,
+    gitHubService,
+    configYmlService,
+    usersService,
+    isomerAdminsService,
+    tokenStore,
+  }: SitesServiceProps) {
+    this.siteRepository = siteRepository
+    this.gitHubService = gitHubService
+    this.configYmlService = configYmlService
+    this.usersService = usersService
+    this.isomerAdminsService = isomerAdminsService
     this.tokenStore = tokenStore
   }
 
   async getBySiteName(siteName: string): Promise<Site | null> {
-    const site = await this.repository.findOne({
+    const site = await this.siteRepository.findOne({
       where: { name: siteName },
     })
 
     return site
   }
 
+  async getSitesForEmailUser(userId: string) {
+    const user = await this.usersService.findSitesByUserId(userId)
+
+    if (!user) {
+      return []
+    }
+
+    return user.site_members.map((site) => site.repo?.name)
```

Comment:
this returns undefined if `repo` doesn't exist right? does the method signature make htis clear

Reply:
ah yea it does, ignore this